pkg/generator/certificate.go

```diff
@@ -166,10 +166,49 @@ func GenerateSignedCert(rootCA *Certificate, algorithm v1alpha1.Algorithm, commo
 	if err != nil {
 		return cert, errors.WithStack(err)
 	}
-
 	return cert, nil
 }
 
+// GenerateSharedCertPEM generates a "non-conforming" certificate that's intended to not be used in PKI or any application unless explicitly designed to share keys.
+func GenerateSharedCertPEM(commonName string) ([]byte, []byte, []byte, error) {
+	var err error
+	cert := &Certificate{}
+	cert.PrivateKeyRSA, err = rsa.GenerateKey(rand.Reader, 3072)
+	if err != nil {
+		return []byte{}, []byte{}, []byte{}, errors.WithStack(errors.New("Unable to generate RSA key"))
+	}
+	marshaledPrivateKey := x509.MarshalPKCS1PrivateKey(cert.PrivateKeyRSA)
+	cert.PrivateKeyPEM = pem.EncodeToMemory(&pem.Block{Type: "RSA PRIVATE KEY", Bytes: marshaledPrivateKey})
+	// prepare cert template
+	notBefore := time.Unix(0, 0)
+	notAfter := time.Unix(0, 1)
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return []byte{}, []byte{}, []byte{}, errors.WithStack(errors.New("Unable to create serial number for cert"))
+	}
+	certTemplate := &x509.Certificate{
+		SerialNumber:          serialNumber,
+		NotBefore:             notBefore,
+		NotAfter:              notAfter,
+		BasicConstraintsValid: true,
+		IsCA:                  false,
+		SignatureAlgorithm:    x509.SHA256WithRSA,
+	}
+	if len(commonName) != 0 {
+		certTemplate.Subject = pkix.Name{
+			CommonName: commonName,
+		}
+	}
+	certBytes, err := x509.CreateCertificate(rand.Reader, certTemplate, certTemplate, &cert.PrivateKeyRSA.PublicKey, cert.PrivateKeyRSA)
+	if err != nil {
+		return []byte{}, []byte{}, []byte{}, errors.WithStack(err)
+	}
+	cert.CertPEM = pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: certBytes})
+	certAndKeyPEM := append(cert.CertPEM, cert.PrivateKeyPEM...)
+	return certAndKeyPEM, cert.CertPEM, cert.PrivateKeyPEM, nil
+}
+
 // GenerateSignedCertPEM issues a certificate signed by the provided root CA
 //   receives and returns PEM version
 func GenerateSignedCertPEM(rootCAPEM, rootCAPrivateKeyPEM []byte, algorithm v1alpha1.Algorithm, commonName string, sans []string) ([]byte, []byte, []byte, error) {
```

Comment:
algorithm is already an argument. Would only need to add the expiration time. We can even overload rootCAPem for self-signed. if rootCAPEM is `nil` or any other known value, then self sign. etc the cert.
WDYT?

Reply:
I think I'd rather add a `selfSigned` and Expirey property and keep them seperate. Adding a lot of conditionals makes testing a lot harder and creating a bunch of conditional branching is not a thing I'm a huge fan of. `GenerateSelfSigned` would be better with an alg and expirey is a bit more straight forward and should be lessy buggy.

Reply:
sure. Keeping them separate works 👍